lib/improver/tests/utilities/test_cube_metadata.py

```diff
@@ -155,7 +261,13 @@ def test_fails_no_points(self):
         cube = create_cube_with_threshold()
         cube.remove_coord(coord_name)
         cube = iris.util.squeeze(cube)
-        changes = {'bounds': [0.1, 2.0], 'units': 'mm'}
+        changes = {
+            'bounds': [0.1, 2.0],
+            'units': {
+                'action': 'set',
```

Comment:
If this is the only use case for "action: set" - the `add_coord` function doesn't use "action" at all as it will only ever need to set the units from scratch.  So I would really recommend losing this option, as it reduces complexity in both the dictionary (no "action" item) and the `update_units` functionality (as it then becomes just a unit conversion and may not even require its own function).

Reply:
I've removed the `action` keyword, so this PR has been greatly simplified to just ensuring that units are converted, if a coordinate is being updated, rather than being set.